File: [6 kyu] WeIrD StRiNg CaSe/main.py
def to_weird_case(words):
    is_upper = True
    result = ""

    for c in words:
        if c == " ":
            result += " "
            is_upper = True
            continue

        result += c.upper() if is_upper else c.lower()
        is_upper = not is_upper

    return result


def to_weird_case(words):
    return " ".join(
        "".join(c.upper() if i % 2 == 0 else c for i, c in enumerate(word.lower()))
        for word in words.split()
    )

File: [6 kyu] WeIrD StRiNg CaSe/main.rs
fn to_weird_case(s: &str) -> String {
    let mut is_upper = true;
    let mut result = String::new();

    for (i, c) in s.chars().enumerate() {
        if c == ' ' {
            result.push(' ');
            is_upper = true;
            continue;
        }

        if is_upper {
            result.push(c.to_ascii_uppercase());
        } else {
            result.push(c.to_ascii_lowercase());
        }
        is_upper = !is_upper;
    }

    result
}
